src/java.desktop/share/classes/javax/print/attribute/standard/OutputBin.java

```diff
@@ -0,0 +1,201 @@
+     */
+    public static final OutputBin BOTTOM = new OutputBin(2);
+
+    /**
+     * The side output bin in the printer.
+     */
+    public static final OutputBin SIDE = new OutputBin(3);
+
+    /**
+     * The left output bin in the printer.
+     */
+    public static final OutputBin LEFT = new OutputBin(4);
+
+    /**
+     * The right output bin in the printer.
+     */
+    public static final OutputBin RIGHT = new OutputBin(5);
+
+    /**
+     * The center output bin in the printer.
+     */
+    public static final OutputBin CENTER = new OutputBin(6);
+
+    /**
+     * The rear output bin in the printer.
+     */
+    public static final OutputBin REAR = new OutputBin(7);
+
+    /**
+     * The face up output bin in the printer.
+     */
+    public static final OutputBin FACE_UP = new OutputBin(8);
+
+    /**
+     * The face down output bin in the printer.
+     */
+    public static final OutputBin FACE_DOWN = new OutputBin(9);
+
+    /**
+     * The large-capacity output bin in the printer.
+     */
+    public static final OutputBin LARGE_CAPACITY = new OutputBin(10);
+
+    /**
+     * Construct a new output bin enumeration value with the given integer
+     * value.
+     *
+     * @param value Integer value
+     */
+    protected OutputBin(int value) {
+        super(value);
+    }
+
+    /**
+     * The string table for class {@code OutputBin}.
+     */
+    private static final String[] myStringTable = {
+            "top",
+            "middle",
+            "bottom",
+            "side",
+            "left",
+            "right",
+            "center",
+            "rear",
+            "face-up",
+            "face-down",
+            "large-capacity",
+    };
+
+    /**
+     * The enumeration value table for class {@code OutputBin}.
+     */
+    private static final OutputBin[] myEnumValueTable = {
+            TOP,
+            MIDDLE,
+            BOTTOM,
+            SIDE,
+            LEFT,
+            RIGHT,
+            CENTER,
+            REAR,
+            FACE_UP,
+            FACE_DOWN,
+            LARGE_CAPACITY,
+    };
+
+    /**
+     * Returns the string table for class {@code OutputBin}.
+     */
+    @Override
+    protected String[] getStringTable() {
+        return myStringTable.clone();
+    }
+
+    /**
+     * Returns the enumeration value table for class {@code OutputBin}.
+     */
+    @Override
+    protected EnumSyntax[] getEnumValueTable() {
+        return (EnumSyntax[]) myEnumValueTable.clone();
+    }
+
+    /**
+     * Get the printing attribute class which is to be used as the "category"
+     * for this printing attribute value.
+     * <p>
+     * For class {@code OutputBin} and any vendor-defined subclasses, the category
+     * is class {@code OutputBin} itself.
+     *
+     * @return printing attribute class (category), an instance of class
+     *         {@link Class java.lang.Class}
+     */
+    @Override
+    public final Class<? extends Attribute> getCategory() {
```

Comment:
> return the supported output-bins when user calls [getSupportedAttributeValues]
Are you sure about that ? Isn't that what lines 834 and 835 of IPPPrintService.java do ?
And one of the supplied tests calls the API you cite .. 

> It does not seem this PR supports
output-bin-default mentioned in the IPP doc, which will be returned via [getDefaultAttributeValue]

I think you are right since I don't see any update to IPPPrintService.getDefaultAttributeValue(..)